### ISSUE TITLE
Compatibility with CommonRLInterface

### ISSUE THREAD

**Issue by rcnlee** [JuliaPOMDP/POMDPModelTools.jl#48]
For context, I am trying to get a POMDPs.MDP environment to work with ReinforcementLearning.jl solvers through CommonRLInterface.jl (https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/issues/181). Specifically, I am trying this for ContinuumWorld and SAC.  At the moment, I am getting the following error:

```julia
> env = convert(RLBase.AbstractEnv, crl_env)

MethodError: no method matching states(::CWorld)
Closest candidates are:
  states(!Matched::POMDPModels.TigerPOMDP) at /Users/rlee18/.julia/packages/POMDPModels/BarLW/src/TigerPOMDPs.jl:10
  states(!Matched::POMDPModels.SimpleGridWorld) at /Users/rlee18/.julia/packages/POMDPModels/BarLW/src/gridworld.jl:29
  states(!Matched::POMDPModels.BabyPOMDP) at /Users/rlee18/.julia/packages/POMDPModels/BarLW/src/CryingBabies.jl:21
  ...

Stacktrace:
 [1] observations(::MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}}) at /Users/rlee18/.julia/packages/POMDPModelTools/tKYU5/src/common_rl/to_env.jl:41
 [2] state_space(::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}, ::Observation{Any}, ::DefaultPlayer) at ./In[45]:1
 [3] state_space(::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}, ::Observation{Any}) at /Users/rlee18/.julia/packages/ReinforcementLearningBase/g77gj/src/interface.jl:459
 [4] state_space(::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}) at /Users/rlee18/.julia/packages/ReinforcementLearningBase/g77gj/src/interface.jl:457
 [5] show(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::MIME{Symbol("text/markdown")}, ::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}) at /Users/rlee18/.julia/packages/ReinforcementLearningBase/g77gj/src/base.jl:97
 [6] show at /Users/rlee18/.julia/packages/ReinforcementLearningBase/g77gj/src/base.jl:79 [inlined]
 [7] limitstringmime(::MIME{Symbol("text/plain")}, ::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}) at /Users/rlee18/.julia/packages/IJulia/IDNmS/src/inline.jl:43
 [8] display_mimestring at /Users/rlee18/.julia/packages/IJulia/IDNmS/src/display.jl:71 [inlined]
 [9] display_dict(::ReinforcementLearningBase.RLBaseEnv{MDPCommonRLEnv{AbstractArray,CWorld,StaticArrays.SArray{Tuple{2},Float64,1,2}},Float32}) at /Users/rlee18/.julia/packages/IJulia/IDNmS/src/display.jl:102
 [10] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [11] invokelatest at ./essentials.jl:709 [inlined]
 [12] execute_request(::ZMQ.Socket, ::IJulia.Msg) at /Users/rlee18/.julia/packages/IJulia/IDNmS/src/execute_request.jl:112
 [13] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [14] invokelatest at ./essentials.jl:709 [inlined]
 [15] eventloop(::ZMQ.Socket) at /Users/rlee18/.julia/packages/IJulia/IDNmS/src/eventloop.jl:8
 [16] (::IJulia.var"#15#18")() at ./task.jl:356
```

(which is not yet implemented).  However, looking at the calling code, I had 2 questions:
1. Why is the current code iterating over ``states(env,m)``?  Is it assuming a discrete space?
2. Some of these functions are optional in both the POMDPs.jl and in CommonRLInterface.jl.  Is there an option to go only with the required interface?

Thanks!

**Comment by zsunberg** [comment]
1. I'm not sure - it looks like in [5] it is running some `show` function from ReinforcementLearningBase. Maybe look there.
2. Indeed we should make it so that `CommonRLInterface.observations` is provided only if there is a valid method for `POMDPs.states`. I will think about how to do this, but it's a ways down in my priority list. Feel free to submit a PR if you have any ideas.

**Comment by zsunberg** [comment]
I think this will be fixed in v0.3.5
